### DISCUSSION TITLE
Build your own firmware

### DISCUSSION THREAD

**Discussion by guntur** [roleoroleo/yi-hack-Allwinner-v2#71] [Q&A]
Hi @roleoroleo 

I want to ask about the compiled source results of mine, why the results are so different from your version.
would you like to check the result?

is there something wrong or missing?

I followed your wiki instructions.

[y21ga_0.1.3-ubuntu-compiled.tgz.zip](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/5830648/y21ga_0.1.3-ubuntu-compiled.tgz.zip)

![compile_yi_hack](https://user-images.githubusercontent.com/3515383/104925941-124d6b80-59d2-11eb-920a-d18cb73d389a.png)

**Comment by roleoroleo** [comment]
I don't know.
All main binaries are missing: framefinder, imggrabber, jq, mosquitto, onvif_srvd, etc...
Probably the compile procedure stopped after busybox.
Check the errors whe you run compile.sh.

**Reply by guntur** [reply]
i didn't notice about error message when compile the source. thanks a lot for reply message.

this is an error message. problem solve after update the library version.

![image](https://user-images.githubusercontent.com/3515383/105137985-cfe97300-5b26-11eb-886d-93a256b36294.png)

![2021-01-19_21-04-59](https://user-images.githubusercontent.com/3515383/105177769-b6155380-5b59-11eb-91e4-2eaf40f80239.png)

And this error message cause by is not enough memory. (find google)

![2021-01-20_6-33-27](https://user-images.githubusercontent.com/3515383/105177806-c6c5c980-5b59-11eb-8793-b75cb3cf6988.png)

I'm using virtual box with ubuntu os, and then I increased memory size from 2 Gb into 4 Gb. 
Finally compiled successfully
I have also tried it on camera and it works
